cas-server-support-osf/src/main/java/org/jasig/cas/support/pac4j/web/flow/ClientAction.java

```diff
@@ -185,6 +190,8 @@ protected Event doExecute(final RequestContext context) throws Exception {
                 final ExternalContext externalContext = ExternalContextHolder.getExternalContext();
                 externalContext.recordResponseComplete();
                 return new Event(this, "stop");
+            } catch (final CredentialsException e) {
+                throw createClientSpecificDelegatedLoginException(client.getClass().getSimpleName(), e);
```

Comment:
Added the missing credential exception which should have been caught and specialized.

Fyi, an example of credential exception for OAuth 2.0 clients (e.g. ORCiD) can be expired/invalid/missing authorization code. another for CAS client (e.g. institutions) can be expired/invalid/missing service ticket.